File: LeetCode/Easy/E_cpp/944. Delete Columns to Make Sorted.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int minDeletionSize(vector<string>& strs) {
        int answer = 0;
        int row = strs.size();
        int col = strs[0].length();

        for (int i = 0; i < col; i++) {
            for (int j = 1; j < row; j++) {
                if (int(strs[j - 1][i]) > int(strs[j][i])) {
                    cout << strs[j - 1][i] << ", " << strs[j][i] << "\n";
                    ++answer;
                    break;
                }
            }
        }

        return answer;
    }
};

File: LeetCode/python/944. Delete Columns to Make Sorted.py
class Solution(object):
    def minDeletionSize(self, strs):
        """
        :type strs: List[str]
        :rtype: int
        """
        answer = 0
        row = len(strs)
        col = len(strs[0])

        for i in range(0, col):
            for j in range(1, row):
                if strs[j - 1][i] > strs[j][i]:
                    answer += 1
                    break

        return answer